ui/desktop/src/extensions.tsx

```diff
@@ -95,7 +94,32 @@ export async function addExtension(
       body: JSON.stringify(config),
     });
 
-    const data = await response.json();
+    const responseText = await response.text();
+
+    if (!response.ok) {
+      const errorMsg = `Server returned ${response.status}: ${response.statusText}. Response: ${responseText}`;
+      console.error(errorMsg);
+      if (toastId) toast.dismiss(toastId);
+      toastError({
+        title: extension.name,
+        msg: 'Failed to add extension',
+        traceback: errorMsg,
+        toastOptions: { autoClose: false },
+      });
+      return response;
+    }
+
+    // Wait a moment for the extension to initialize
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
i thought the api for starting an extension only respond when the extension either succeeds or fails? what is this wait doing?

Reply:
this is a remanent from debugging... was getting 428's and thought it wasn't waiting long enough